### DISCUSSION TITLE
Any chance of releasing `future_into_py_futlike`?

### DISCUSSION THREAD

**Discussion by winstxnhdw** [emmett-framework/granian#433] [General]
Was wondering if you would release `future_into_py_futlike` as a crate.

I've been using `pyo3-async-runtimes` to return async functions from Rust, and I found it extremely slow. About 20x slower than running my GIL-less Rust function within a Python thread and using `wrap_futures` on it. I haven't tested it yet but I assume `future_into_py_futlike` wouldn't have such an issue.

**Comment by gi0baro** [comment]
So, the issue with `future_into_py_futlike` is that at the moment is quite tightened to some specific Granian runtime characteristics, so it would be quite useless to just put that into a crate without the rest of the runtime.

In terms of general functionality, that function won't make anything super-special, it returns a `pyclass` which acts as an `asyncio.Future`. I think something like the windows implementation can be generalised more easily 
https://github.com/emmett-framework/granian/blob/a6cfc2f1f9d94ae7f57d6400cfee071eac328aa5/src/runtime.rs#L221-L271
as it depends only on a _glue class_ 
https://github.com/emmett-framework/granian/blob/a6cfc2f1f9d94ae7f57d6400cfee071eac328aa5/src/callbacks.rs#L260-L276
and the performance difference shouldn't be so noticeable in 99% of use cases.

Given the amount of lines and still the hard dependency on tokio, I'm not sure still wether a crate would be a good idea for that..
What are your thoughts @winstxnhdw?

**Reply by winstxnhdw** [reply]
I definitely believe it’ll be incredibly helpful to everyone if it was a crate. The current ecosystem for returning futures from Rust is incredibly lacking. 

I can’t figure out how you are spawning a new thread with the Granian runtime as you showed in the Windows example.

**Reply by winstxnhdw** [reply]
I came back to look at this and tried integrating your solution, but I am struggling to avoid cloning whatever data from Python inside `future_into_py_futlike`.
